src/command_modules/azure-cli-redis/azure/cli/command_modules/redis/_params.py

```diff
@@ -3,43 +3,64 @@
-                   completer=get_resource_name_completion_list('Microsoft.Cache/redis'))
+        cache_name = CLIArgumentType(options_list=['--name', '-n'], help='Name of the Redis cache.', id_part='name',
+                                     completer=get_resource_name_completion_list('Microsoft.Cache/redis'))
+        cache_name_without_id_part = CLIArgumentType(overrides=cache_name, id_part=None)
+        # pylint: disable=line-too-long
+        format_type = CLIArgumentType(options_list=['--file-format'], help='Format of the blob (Currently rdb is the only supported format, with other formats expected in the future)')
+
+        c.argument('name', arg_type=cache_name)
         c.argument('redis_configuration', help='JSON encoded configuration settings. Use @{file} to load from a file.',
                    type=JsonString)
         c.argument('reboot_type', arg_type=get_enum_type(RebootType))
         c.argument('key_type', arg_type=get_enum_type(RedisKeyType))
+        c.argument('files', help='SAS url for blobs that needs to be imported', nargs='+')
+        c.argument('format', arg_type=format_type)
+        c.argument('file_format', arg_type=format_type)
+        c.argument('container', help='SAS url for container where data needs to be exported to')
+        c.argument('prefix', help='Prefix to use for exported files')
+        c.argument('cache_name', arg_type=cache_name)
+        c.argument('shard_count', type=int, help='The number of shards to be created on a Premium Cluster Cache.')
+        # pylint: disable=line-too-long
+        c.argument('subnet_id', help='The full resource ID of a subnet in a virtual network to deploy the redis cache in. Example format /subscriptions/{subid}/resourceGroups/{resourceGroupName}/Microsoft.{Network|ClassicNetwork}/VirtualNetworks/vnet1/subnets/subnet1')
+        c.argument('static_ip', help='Required when deploying a redis cache inside an existing Azure Virtual Network')
+        c.argument('tenant_settings', type=JsonString, help='Json dictionary with tenant settings')
+        # pylint: disable=line-too-long
+        c.argument('tags', type=JsonString, help='Json dictionary with Resource tags. Example : {\"testKey\":\"testValue\"}"""')
+        # pylint: disable=line-too-long
+        c.argument('zones', type=shell_safe_json_parse, help='A list of availability zones denoting where the resource needs to come from')
         c.argument('shard_id', type=int)
         c.argument('sku', help='Type of Redis cache.', arg_type=get_enum_type(SkuName))
-        c.argument('vm_size', help='Size of Redis cache to deploy. Example : values for C family (C0, C1, C2, C3, C4, '
-                                   'C5, C6). For P family (P1, P2, P3, P4)')
+        # pylint: disable=line-too-long
+        c.argument('minimum_tls_version', help='Specifies the TLS version required by clients to connect to cache', arg_type=get_enum_type(TlsVersion))
+        # pylint: disable=line-too-long
```

Comment:
Since this is a params.py file, just put this at the top after the license header.  THen you only need it once.